File: app/src/main/java/com/example/expensetrcaker/data/WalletEntity.java
package com.example.expensetrcaker.data;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "wallets")
public class WalletEntity {
    @PrimaryKey(autoGenerate = true)
    private long id;
    private String name;
    private double balance;
    private String bank;
    private int colorStart;
    private int colorEnd;

    public WalletEntity(String name, double balance, String bank, int colorStart, int colorEnd) {
        this.name = name;
        this.balance = balance;
        this.bank = bank;
        this.colorStart = colorStart;
        this.colorEnd = colorEnd;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getBalance() {
        return balance;
    }

    public void setBalance(double balance) {
        this.balance = balance;
    }

    public String getBank() {
        return bank;
    }

    public void setBank(String bank) {
        this.bank = bank;
    }

    public int getColorStart() {
        return colorStart;
    }

    public void setColorStart(int colorStart) {
        this.colorStart = colorStart;
    }

    public int getColorEnd() {
        return colorEnd;
    }

    public void setColorEnd(int colorEnd) {
        this.colorEnd = colorEnd;
    }
}

File: app/src/main/java/com/example/expensetrcaker/data/WalletEntity.kt
package com.example.expensetrcaker.data

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Entity(tableName = "wallets")
data class WalletEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val balance: Double,
    val bank: String,
    val colorStart: Int, // Store as Int ARGB
    val colorEnd: Int
)

@Dao
interface WalletDao {
    @Query("SELECT * FROM wallets")
    fun getAllWallets(): Flow<List<WalletEntity>>

    @Insert
    suspend fun insertWallet(wallet: WalletEntity)

    @Update
    suspend fun updateWallet(wallet: WalletEntity)

    @Query("UPDATE wallets SET balance = balance + :amount WHERE id = :walletId")
    suspend fun updateBalance(walletId: Long, amount: Double)

    @Delete
    suspend fun deleteWallet(wallet: WalletEntity)
}
